LanguageFeatures/Extension-types/static_analysis_extension_types_A33_t01.dart

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion An extension type V is always-exhaustive if and only if its
+/// instantiated representation type is always-exhaustive.
+///
+/// @description Check that an extension type is always-exhaustive if its
+/// representation type is always-exhaustive.
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=inline-class
+
+import '../../Utils/expect.dart';
+
+extension type BoolET(bool b) {}
+
+String test1(BoolET b) {
+  switch (b) {
+    case BoolET(b: true):
+      return "True";
+    case BoolET(b: false):
+      return "False";
+  }
+}
+
+String test2(BoolET b) =>
+    switch (b) {
+      BoolET(b: true) => "True",
+      BoolET(b: false) => "False"
+    };
```

Comment:
The notion of `BoolET` being always-exhaustive doesn't matter here: Switch expressions must be exhaustive no matter what they're switching on.

One could perhaps claim that this is a test of the exhaustiveness analysis: It is only possible for for the exhaustiveness analysis to conclude that this switch expression is exhaustive if it uses the extension type erasure to discover that the scrutinee is a `bool` and the cases cover `true` and `false`.

But in that case the description should say something like "testing that the exhaustiveness analysis uses extension type erasure to compute space unions of the matched value type and the patterns". But that might be covered by some other test already? (constant only, perhaps?)

Reply:
Yes, we have such test for constants. Dropped test for switch expression